packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -367,31 +371,54 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
         );
     }
 
+    /**
+     * Set up a mapping in the redirect table from fromId to toId. Also, notify the runtime that a reference is added
+     * which is required for GC.
+     */
+    private setRedirection(fromId: string, toId: string | undefined) {
+        this.redirectTable.set(fromId, toId);
+        // Notify runtime of a reference added if toId is not undefined. It can be undefined when a blob is uploaded in
+        // detached mode. In this case, the entry will be updated when the blob is updated.
+        if (toId !== undefined) {
+            this.addedBlobReference(this.getBlobGCNodePath(fromId), this.getBlobGCNodePath(toId));
+        }
+    }
+
     private deleteAndEmitsIfEmpty(id: string) {
-        this.pendingBlobs.delete(id);
-        if (!this.hasPendingBlobs) {
-            this.emit("noPendingBlobs");
+        if (this.pendingBlobs.has(id)) {
```

Comment:
Nit: 
`if (!this.pendingBlobs.has(id)) { return; }`
and 
`if (this.hasPendingBlobs) { return; }`
Feel free to ignore